src/Widgets/AppSettingsView.vala

```diff
@@ -101,7 +88,7 @@ public class Widgets.AppSettingsView : Gtk.Grid {
     }
 
     private void update_header () {
-        header.set_title (selected_app.app_info.get_display_name ());
-        header.set_icon (selected_app.app_info.get_icon ());
+        title = selected_app.app_info.get_display_name ();
+        icon_name = selected_app.app_info.get_icon ().to_string ();
```

Comment:
```suggestion
        ((Gtk.Image) display_widget).gicon = selected_app.app_info.get_icon ();
```
And in `construct {}` do:
display_widget = new Gtk.Image ();
((Gtk.Image) display_widget).pixel_size = 32;

Reply:
So it turns out that display_widget is construct only and SimpleSettingsPage doesn't do anything with it at all. So it seems like some work is needed in granite to fix this :/